src/agent0/core/hyperdrive/interactive/hyperdrive_test.py

```diff
@@ -268,3 +269,85 @@ def test_no_policy_call(fast_chain_fixture: LocalChain):
         hyperdrive_agent.execute_policy_action()
 
     remote_chain.cleanup()
+
+
+@pytest.mark.anvil
+def test_no_approval(fast_chain_fixture: LocalChain):
+    """Test no approval error on the base remote chain."""
+    initial_pool_config = LocalHyperdrive.Config()
+    local_hyperdrive = LocalHyperdrive(fast_chain_fixture, initial_pool_config)
+    private_key = make_private_key()
+
+    # Initialize the agent here to fund
+    fast_chain_fixture.init_agent(
+        base=FixedPoint(1_000_000), eth=FixedPoint(1_000), private_key=private_key, pool=local_hyperdrive
+    )
+    hyperdrive_addresses = local_hyperdrive.hyperdrive_address
+    # Connect to the local chain using the remote hyperdrive interface
+    # To avoid a port conflict with the existing db container in `fast_chain_fixture`,
+    # we use a separate chain port here
+
+    remote_chain = Chain(
+        fast_chain_fixture.rpc_uri,
+        Chain.Config(
+            db_port=40000,
+        ),
+    )
+    interactive_remote_hyperdrive = Hyperdrive(remote_chain, hyperdrive_addresses)
+    # Create agent with same private key
+    hyperdrive_agent = remote_chain.init_agent(private_key=private_key, pool=interactive_remote_hyperdrive)
+
+    # Make a call without approval
+    try:
+        hyperdrive_agent.add_liquidity(base=FixedPoint(1_000))
+    except ContractCallException as exc:
+        assert "Insufficient allowance: " in exc.args[0]
+
+    try:
+        hyperdrive_agent.open_long(base=FixedPoint(1_000))
+    except ContractCallException as exc:
+        assert "Insufficient allowance: " in exc.args[0]
+
+    try:
+        hyperdrive_agent.open_short(bonds=FixedPoint(1_000))
+    except ContractCallException as exc:
+        assert "Insufficient allowance: " in exc.args[0]
+
+    remote_chain.cleanup()
+
+
+@pytest.mark.anvil
+def test_out_of_gas(fast_chain_fixture: LocalChain):
```

Comment:
can you add a test or modify this one to also include a case where there is enough gas?

Reply:
aka "happy path" test

Reply:
We have plenty of other tests that do trades without an explicit gas, which all pass